### ISSUE TITLE
Do not delete route when adding an additional destination

### ISSUE THREAD

**Issue by TheSnoozer** [osmandapp/OsmAnd#15382]
# 🚀 feature request

### Description

Hello,
thanks for the great app. I have a small detail regarding "adding an additional destination".
I checked if i could find a similar issue, but haven't found anything.
Let's assume we navigate to point A and the OSMAND app already calculated the route to that point. When now now selecting yet another location we want to travel to (point B) and the dialoge with "you already selected a destination" pops up, selecting the option "add as an additional destination" will wipe the entire route calculation.

IMHO there is no need to wipe the already calculated route to point A since it will be the first target on the desired route. 

### Describe the solution you'd like
OSMAND should only calculate the route A -> B, but not current location -> A.

### Describe alternatives you've considered
Calculate the route beforehand, which doesn't work with long routes.

**Comment by vshcherb** [comment]
Makes sense, I think there is a speedup in the route calculation but the route is not displayed until full route is calculated.

**Comment by TheSnoozer** [comment]
Thanks for your insight! Indeed that makes sense too! 

PS: I'm using that approach as workaround for https://github.com/osmandapp/OsmAnd/issues/14896. Effectively for very long trips I have multiple waypoints (each 200-300km) on the desired route. So when getting close to the "next" waypoint I want to add the next one but would like to have the already calculated route visible (so I don't take a wrong turn somewhere). For now the best way is to just take a quick stop and have the route calculation finish.

**Comment by tradavyvy** [comment]
Better than 'nice to have'
